File: src/main/java/com/terminal/controller/DepartamentosController.java
package com.terminal.controller;

import com.terminal.entity.Departamentos;
import com.terminal.service.DepartamentosService;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/departamentos")
@CrossOrigin(origins = "*") // Permite peticiones desde el frontend
public class DepartamentosController {

    private final DepartamentosService service;

    public DepartamentosController(DepartamentosService service) {
        this.service = service;
    }

    @GetMapping
    public List<Departamentos> getDepartamentos() {
        return service.getAll();
    }
}

File: src/departamentos/departamentos.controller.ts
import { Controller, Get } from '@nestjs/common';
import { DepartamentosService } from './departamentos.service';

@Controller('departamentos')
export class DepartamentosController {
  constructor(private readonly departamentosService: DepartamentosService) {}

  @Get()
  getDepartamentos() {
    return this.departamentosService.getDepartamentos();
  }
}
